### PR TITLE
VcfAncestorAnnotation in selectionMCMC

### PR DESCRIPTION
**Pull request by rimangan**
For our selection model, we would like to "polarize" Allele Frequency Spectra, or to set the "ancestral" allele as the ref allele and the "derived" allele as the alt as the math underlying our model is based on derived allele frequency spectra. With the changes here, we can now filter VCF files for alleles with ancestral annotation, generate polarized AFS structs based on these ancestral allele annotations, and feed these polarized AFS structs into the MCMC model. Several additional tests have been added to cover the new functionality. Passes tests.

### CODE REVIEW

**Comment by DanielSnellings** [comment]
Could you remove the lift stuff from this PR. I think it's covered in another PR.

**Review by craiglowe — APPROVED** [review]
Wonderful!  Great to see all of these tools and excited to see what you get from the mcmc with derived alleles.

File: popgen/stationarity.go [hunk 1]
```diff
@@ -75,6 +79,16 @@ func VcfToAFS(filename string) AFS {
 				}
 			}
 			if currentSeg.n != 0 { //catches variants where there is no data from the samples (can happen when filtering columns)
```
[the remark this answers is not in the document]
Comment (rimangan) [reply popgen/stationarity.go:83]: we could handle this a couple of different ways. I suppose one option is we fatal here and demand the user provide vcf data that has been filtered for segregating sites. The other is that they silently get dropped, but that could cause issues. Do you have a particular opinion?

File: cmd/selectionMCMC/selectionMCMC.go [hunk 2]
```diff
@@ -8,16 +8,20 @@ import (
 	"log"
 )
 
-func selectionMCMC(filename string, outFile string, muZero float64, sigmaZero float64, iterations int, randSeed bool, setSeed int64) {
+func selectionMCMC(filename string, outFile string, muZero float64, sigmaZero float64, iterations int, randSeed bool, setSeed int64, unPolarized bool) {
 	common.RngSeed(randSeed, setSeed)
-	data := popgen.VcfToAFS(filename)
-	popgen.MetropolisHastings(data, muZero, sigmaZero, iterations, outFile)
+	data, err := popgen.VcfToAFS(filename, !unPolarized) //VcfToAFS is writted with polarized as the argument for clarity, so the bool is flipped here.
+	if err != nil {
```
Comment (craiglowe) [inline cmd/selectionMCMC/selectionMCMC.go:14]: I think there is an error handling package now that will shorten this to one line.

File: cmd/vcfAFS/vcfAFS.go [hunk 3]
```diff
@@ -3,14 +3,18 @@ package main
...
 	"github.com/vertgenlab/gonomics/popgen"
 	"log"
 )
 
-func vcfAFS(vcfFile string, outFile string) {
-	g := popgen.VcfToAFS(vcfFile)
-	f := popgen.AFSToFrequency(g)
+func vcfAFS(vcfFile string, outFile string, unPolarized bool) {
+	g, err := popgen.VcfToAFS(vcfFile, !unPolarized) //VcfToAFS is written in terms of polarized, so this is inverted here.
+	if err != nil {
```
Comment (craiglowe) [inline cmd/vcfAFS/vcfAFS.go:14]: error handing package could be used here as well

File: cmd/vcfFilter/vcfFilter_test.go [hunk 4]
```diff
@@ -8,26 +8,30 @@ import (
...
+	{"testdata/test.vcf", "testdata/expectedOut.vcf", "testdata/test.group", "chr3", 10, 1000, 0.0, "", "", true, true, true, false, false},
+	{"testdata/test_removeNoAncestor.vcf", "testdata/expected_removeNoAncestor.vcf", "", "", 0, 100, 0.0, "", "", false, false, false, true, false},
+	{"testdata/test_onlyPolarizable.vcf", "testdata/expected_onlyPolarizable.vcf", "", "", 0, 100, 0.0, "", "", false, false, false, false, true},
 }
 
 func TestVcfFilter(t *testing.T) {
 	var err error
 	for _, v := range VcfFilterTests {
-		vcfFilter(v.inputFile, "tmp.vcf", v.groupFile, v.chrom, v.minPos, v.maxPos, v.ref, v.alt, v.minQual, v.biAllelicOnly, v.substitutionsOnly, v.segregatingSitesOnly)
+		vcfFilter(v.inputFile, "tmp.vcf", v.groupFile, v.chrom, v.minPos, v.maxPos, v.ref, v.alt, v.minQual, v.biAllelicOnly, v.substitutionsOnly, v.segregatingSitesOnly, v.removeNoAncestor, v.onlyPolarizableAncestors)
```
Comment (craiglowe) [inline cmd/vcfFilter/vcfFilter_test.go:34]: Should this be testdata/tmp.vcf?

File: popgen/stationarity.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (craiglowe) [reply popgen/stationarity.go:91]: Thank you for adding the checks above.  Can we remove this 'if' now that we have those checks?

**Review by craiglowe — APPROVED** [review]
Looks great!  I had a couple comments, but nothing to stop merging this in.

**Review by RavenYL — DISMISSED** [review]

File: popgen/stationarity.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (rimangan) [reply popgen/stationarity.go:91]: great catch thanks

File: cmd/vcfFilter/vcfFilter_test.go (hunk 4 above)
Comment (rimangan) [reply cmd/vcfFilter/vcfFilter_test.go:34]: It gets deleted shortly after so either way, perhaps putting it in testdata is more consistent but it's only present when the program is testing

**Review by craiglowe — APPROVED** [review]
Great work!  Thank you for taking all the comments into consideration

**Review by cfauci — APPROVED** [review]

**Review by DanielSnellings — APPROVED** [review]
